packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp

```diff
@@ -1813,6 +1813,14 @@ namespace Tpetra {
                   Teuchos::ArrayRCP<const LocalOrdinal>& columnIndices,
                   Teuchos::ArrayRCP<const Scalar>& values) const;
 
+    /// Gets just the values array.  This *will* be a shallow copy
+    /// of the array (at least on the host memory space.  This
+    /// is not a const function, since the user can change these
+    /// values.
+    ///
+    /// \param values [out] Array of values. 
+    void getAllValues(Teuchos::ArrayRCP<Scalar>& values);
```

Comment:
Why is this needed? Couldn't you just use `getAllValues` and ignore rowptr and column indices?

Reply:
In the Epetra case that actually copies the rowptr (because int vs. size_t)